### ISSUE TITLE
💡 Feature Request - Private DNS Zones for Private Links in a separate Resource Group

### ISSUE THREAD

**Issue by cloudchristoph** [Azure/ALZ-Bicep#164]
## Describe the solution you'd like

I recommend moving the Private DNS Zones for Private Link scenarios to a separate Resource Group.

In actual customer environments, it has been found that DNS management is often managed administratively differently than firewall or hub network management.

Furthermore, Azure RBAC rules that affect DNS records are much easier to control if the collected DNS zones are kept in their own resource group. Also any Azure Policies (as described in Issue #137) can be assigned more explicitly (from my point of view).

In addition, this change results in a much clearer resource group for the hub network.

## Describe alternatives you've considered

To solve it, we did two things:
1. We rolled out the "hubNetworking" module with private DNS zones disabled (to avoid breaking the "API" so to speak - we didn't change the original `hubNetworking.bicep` module).
    
    Example from our Deployment Bicep file:
    ```bicep
    module hubNetworking '../infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep' = {
      scope: hubNetworkingRg
      name: 'hubNetworking'
      params: {
        parRegion: location
        parAzureFirewallEnabled: true
        parAzureFirewallTier: 'Standard'
        parCompanyPrefix: 'swo'
        parPrivateDNSZonesEnabled: false
        parExpressRouteGatewayConfig: {}
      }
    }
    
    ```

2. We implemented another module "privateDnsZones", which is essentially a copy of "hubNetworking" in terms of the private DNS zones (comes after this report as a pull request), and rolled it out as follows:
    
    ```bicep
    module privateDnsZones '../infra-as-code/bicep/modules/privateDnsZones/privateDnsZones.bicep' = {
      scope: privateDnsZoneRg
      name: 'privateDnsZones'
      params: {
        parRegion: location
        parPrivateDNSZonesEnabled: true
        parHubVirtualNetworkId: hubNetworking.outputs.outHubVirtualNetworkID
      }
    }
    ```

## Additional context

Our whole deployment Bicep file (placed at `./deployment/hubNetworking.bicep`) for this environment looks like this:

```bicep
// Deploy Hub Network in SWON environment

targetScope = 'subscription'

param location string = 'westeurope'

resource hubNetworkingRg 'Microsoft.Resources/resourceGroups@2021-04-01' = {
  name: 'rg-core-network-weu-01'
  location: location
}

resource privateDnsZoneRg 'Microsoft.Resources/resourceGroups@2021-04-01' = {
  name: 'rg-privatednszones-weu-01'
  location: location
}

module hubNetworking '../infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep' = {
  scope: hubNetworkingRg
  name: 'hubNetworking'
  params: {
    parRegion: location
    parAzureFirewallEnabled: true
    parAzureFirewallTier: 'Standard'
    parCompanyPrefix: 'swo'
    parDDoSEnabled: false
    parPrivateDNSZonesEnabled: false
    parExpressRouteGatewayConfig: {}
  }
}

module privateDnsZones '../infra-as-code/bicep/modules/privateDnsZones/privateDnsZones.bicep' = {
  scope: privateDnsZoneRg
  name: 'privateDnsZones'
  params: {
    parRegion: location
    parPrivateDNSZonesEnabled: true
    parHubVirtualNetworkId: hubNetworking.outputs.outHubVirtualNetworkID
  }
}
```

**Comment by cloudchristoph** [comment]
To be honest: I would suggest to remove the Private DNS Zones completely from the hubNetworking module/resource group for the reasons stated here.

The PR (#165) is just a "step" in this direction. 

As the authors of the module: What are your thoughts about this, @jtracey93 @aultt?

**Comment by jtracey93** [comment]
Linking to #137

Will review early in the week

**Comment by jtracey93** [comment]
Just to update everyone here. Have assigned this to @ejhenry and also the associated PR #165 to review and oversee.

As discussed by the ALZ-Bicep core team today. We propose the following.

1. Split out Private DNS Zones to a separate module
2. Still call this module from Hub Networking and deploy to the default resource group
   - As this is parity across all the ALZ implementations (Terraform, Portal etc.) 
3. Investigate if we can easily parameterize the resource group to deploy the private DNS zones to inside of hub networking and default to the same resource group as it is today for parity reasons (as above) but also provide flexibility

This is also related to #137 as the same code base here is important for the outputs to be used in the policy assignment 👍

@cloudchristoph expect to hear from @ejhenry in the coming days and maybe some possible tweaks to your PR to help here, if you want/can (time wise). If not we may close your PR and develop ourselves. But I'll leave that for you and @ejhenry to work out 👍👍

**Comment by cloudchristoph** [comment]
Implemented with #165
